meta-lmp-base/recipes-containers/docker/files/0001-layer-ensure-layer-files-are-synced-to-disk.patch

```diff
@@ -0,0 +1,55 @@
+From 10d9e53a64636bcaafcfdb05f3412631197cca81 Mon Sep 17 00:00:00 2001
+From: Mike Sul <mike.sul@foundries.io>
+Date: Fri, 15 Sep 2023 21:27:51 +0200
+Subject: [PATCH] layer: Ensure layer files are synced to disk
+
+Invoke the sync system call to guarantee the proper saving of extracted
+layer files on disk before committing the layer. This safeguards against
+potential issues where the Docker daemon may erroneously consider a layer
+as committed when not all its files are correctly stored on disk.
+Such issues could result in empty or incomplete layer files, particularly
+if the daemon is abruptly interrupted during the layer injection process
+into the Docker store. As a result, software running in the container
+based on the layer with 'broken' files will behave incorrectly.
+
+Invoking the sync system call is crucial to prevent this issue,
+as the incorrect behavior of software running inside a container
+with 'broken' files is often not immediately apparent.
+Container engineers may begin debugging the container's software, unaware
+that the root cause lies in the fact that some of the container files are
+simply 'broken'.
+
```

Comment:
can please add the patch status?

```
Upstream-Status: Pending
```
until it is not subited upstream.

Reply:
Done. However, I am not sure moby will accept this patch as it is because (citation from https://github.com/moby/moby/pull/45900#issuecomment-1629672564):

```
We would not be able to accept a PR which uses syscall.Sync() as it is such a blunt instrument which could have a severe performance impact on workloads in production servers with big fs caches and uninterruptible power supplies, but a more targeted fsync of just the unpacked files and directories may be more palatable.
```

I kind of agree with him because doing `sync` for each layer write on a big servers with huge disks and lots of volumes supposedly will have a huge impact on performance.

I've already tried and checked doing `fsync` on each file as he suggested. However, it causes significant performance degradation on a real device (rpi3), layer extraction takes significantly more time.

I'll create a jira ticket for this one and do another fight round for this issue, maybe I'll come up with something that makes both embedded and data center servers worlds happy.

Reply:
https://foundriesio.atlassian.net/browse/FFTK-2815

Reply:
Understant and agree with you as it is very specific for our use case, maybe `Upstream-Status: Inappropriate [lmp specific]` is better there.